### ISSUE TITLE
options field in the ssh_keys hash ignored? 

### ISSUE THREAD

**Issue by Falkor** [camptocamp/puppet-accounts#15]
Hello, 

I cannot find the proper way to have the `camptocamp/accounts` module taking into account the `options` field elligible in the `ssh_authorized_keys` resource. 

For instance, using the following Hiera configuration: 

``` puppet
accounts::users:
  localadmin:
    comment: Local Admin
    home: /var/lib/localadmin
# SSH keys 
#
accounts::ssh_keys:
  svarrette@machine.domain.com:
    type: ssh-dss
    public: AAAAB...
    options: "environment=\"SYSADMIN_USER=svarrette\" "
  svarrette@workstation:
    type: ssh-rsa
    public: 5re...
    options: "environment=\"SYSADMIN_USER=svarrette\" "
  jdoe:
    type: ssh-dss
    public: MAAACBAKQMf834bHh4TFMecBKK...
    options: "environment=\"SYSADMIN_USER=jdoe\" "
```

and applying the following manifest: 

``` puppet
node default {
    class { 'accounts':
        users          => hiera_hash('accounts::users', {}),
        ssh_keys       => hiera_hash('accounts::ssh_keys', {}),
        purge_ssh_keys => true,
    }

    accounts::account{ 'localadmin' :
        authorized_keys => [ 'svarrette@machine.domain.com',
                             'svarrette@workstation',
                             'jdoe'
                             ]
    }
}
```

... lead to the creation of the `localadmin` account, yet the created authorized_keys seems to have ignored the `options` field since it contains: 

```
    $> cat /var/lib/localadmin/.ssh/authorized_keys
   # HEADER: This file was autogenerated at 2015-05-30 15:19:12 +0000
   # HEADER: by puppet.  While it can still be managed manually, it
   # HEADER: is definitely not recommended.
   ssh-dss AAAAB... svarrette@machine.domain.com-on-localadmin
   ssh-dss MAAACBAKQMf834bHh4TFMecBKK... jdoe-on-localadmin
   ssh-rsa 5re... svarrette@workstation-on-localadmin
```

while I cas expecting of course: 

```
   environment="SYSADMIN_USER=svarrette"  ssh-dss AAAAB... svarrette@machine.domain.com-on-localadmin
    environment="SYSADMIN_USER=jdoe" ssh-dss MAAACBAKQMf834bHh4TFMecBKK... jdoe-on-localadmin
    environment="SYSADMIN_USER=svarrette" ssh-rsa 5re... svarrette@workstation-on-localadmin
```

I probably missed something, unless it's a bug on the way the `options` field is treated.
Any suggestion?

**Comment by jardleex** [comment]
Hello @Falkor,
I can confirm this behavior on Ubuntu 16.04 with Puppet 3.8.5.
This patch may solve the problem:
```
diff --git a/manifests/authorized_key.pp b/manifests/authorized_key.pp
index c51b9d5..f9439fa 100644
--- a/manifests/authorized_key.pp
+++ b/manifests/authorized_key.pp
@@ -41,7 +41,7 @@ define accounts::authorized_key(
       ssh_authorized_key { $_ssh_authorized_key_title:
         ensure  => $ensure,
         key     => $::accounts::ssh_keys[$ssh_key]['public'],
-        options => $options,
+        options => $::accounts::ssh_keys[$ssh_key]['options'],
         target  => $_target,
         type    => $::accounts::ssh_keys[$ssh_key]['type'],
         user    => $_user,
```

But I'm not fully sure if this will cover all use cases of the options parameter.
Maybe @mcanevet or @raphink can take a look and verify this?

Best

Jard

**Comment by Falkor** [comment]
any update using the above patch ?

**Comment by mcanevet** [comment]
If someone provides a PR with unit test proving the bug, and the patch that fixes the bug I'd gladly merge it.

**Comment by Falkor** [comment]
I tried to prepare one -- see https://github.com/ULHPC/puppet-accounts/blob/bugfix/issue15/manifests/authorized_key.pp

However as @jadleex feared, it does not seem to permit to sustain the environment option with errors like: 

``` 
Error: Failed to apply catalog: Parameter options failed on Ssh_authorized_key[XXX]: Option environment="SYSADMIN_USER=svarrette"  is not valid. A single option must either be of the form 'option' or 'option="value". Multiple options must be provided as an array at /etc/puppetlabs/code/environments/svarrette/modules/accounts/manifests/authorized_key.pp:41
```

Once I manage to solve this issue, I'll make you a proper PR over this branch

**Comment by raphink** [comment]
@Falkor 	were you able to fix the issue?

**Comment by raphink** [comment]
@Falkor does #56 fix it for you?

**Comment by jardleex** [comment]
I can confirm that #56 accepts single arguments as well as a list of them as written in the PR.

**Comment by raphink** [comment]
Fixed in #56
